src/App.tsx

```
@@ -8,53 +8,185 @@ import { PostsList } from './components/PostsList';
+  const [errorComments, setErrorComments] = useState(false);
+
+  const [isUserSelected, setIsUserSelected] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  const [comments, setComments] = useState<Comment[]>([]);
+
+  async function fetchCreateComment({
+    name,
+    email,
+    body,
+    postId,
+  }: Omit<Comment, 'id'>) {
+    try {
+      const result = await createComments({ name, email, body, postId });
+
+      setComments(currentComments => [...currentComments, result]);
+    } catch (errorCreatComments) {
+      throw errorCreatComments;
+    }
+  }
+
+  async function fetchPosts(userId: number) {
+    setIsLoading(true);
+    setError(false);
+    setIsUserSelected(!!userId);
+
+    if (userId !== selectedUserId) {
+      setIsSidebarOpen(false);
+      setSelectedUserId(userId);
+    }
+
+    try {
+      const results = await getPosts(userId);
+
+      setPosts(results);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  async function fetchComments(postId: number) {
+    setIsLoadingComments(true);
+    setErrorComments(false);
+
+    try {
+      const results = await getComments(postId);
+
+      setComments(results);
+    } catch {
+      setErrorComments(true);
+    } finally {
+      setIsLoadingComments(false);
+    }
+  }
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  async function fetchDeleteComment(id: number) {
+    setComments(currentComments =>
+      currentComments.filter(comment => comment.id !== id),
+    );
+
+    try {
+      await deleteComments(id);
+    } catch (errDelete) {
+      throw errDelete;
+    }
+  }
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const loadingUsers = await getUsers();
+
+        setUsers(loadingUsers);
+      } catch (errUsers) {
```

The fetchUsers catch block rethrows the error (`throw errUsers;`) from inside useEffect. Throwing from an async function launched in useEffect will result in an unhandled rejection. This violates the previous review's instruction: "Replace empty catch blocks with at least `console.error(error)` and, where appropriate, set error state." Replace the rethrow with `console.error(errUsers)` and consider setting a top-level error state so the UI can show a notification or fallback behavior when initial users load fails.